badge_server/main.py

```diff
@@ -75,519 +88,235 @@ def _get_pair_status_for_packages(pkg_sets):
+        dep_detail['priority'] = dep.priority.level.name
+        dep_detail['detail'] = dep.priority.details
+        outdated_depencdency_name_to_details[dep.name] = dep_detail
+    result_dict['status'] = max_level.name
+    result_dict['details'] = outdated_depencdency_name_to_details
+    result_dict['deprecated_deps'] = deprecated_deps
+
+    return result_dict
+
+
+def _get_badge_status(
+        self_compat_res: dict,
+        google_compat_res: dict,
+        dependency_res: dict) -> str:
+    """Get the badge status.
+
+    The badge status will determine the right hand text and the color of
+    the badge.
+
+    Badge status to color mapping:
+    SUCCESS -> green
+    UNKNOWN -> purple
+    CHECK_WARNING -> red
+
+    See badge_utils.STATUS_COLOR_MAPPING.
+    """
+    dep_status = dependency_res['status']
+    dep_status = 'SUCCESS' if dep_status == 'UP_TO_DATE' else dep_status
 
-    response = flask.make_response(
-        pybadges.badge(
-            left_text=badge_name,
-            right_text=status,
-            right_color=color,
-            whole_link=details_link))
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
-    response.headers['Cache-Control'] = 'no-cache'
-    response.add_etag()
+    _statuses = []
+    for pyver in ['py2', 'py3']:
+        _statuses.append(self_compat_res[pyver]['status'])
+        _statuses.append(google_compat_res[pyver]['status'])
+    _statuses.append(dep_status)
 
-    return response
+    if ['SUCCESS'] * len(_statuses) == _statuses:
+        return 'SUCCESS'
+    elif 'UNKNOWN' in _statuses:
+        return 'UNKNOWN'
+    return 'CHECK_WARNING'
 
 
-@app.route('/one_badge_target')
-def one_badge_target():
-    package_name = flask.request.args.get('package')
-    commit_number = badge_utils._calculate_commit_number(package_name)
+def _get_check_results(package_name: str, commit_number: str = None):
+    """Gets the compatibility and dependency check results.
 
-    status, _, self_compat_res, google_compat_res, dependency_res = \
-        _get_all_results_from_cache(package_name, commit_number)
+    Returns a 3 tuple: self compatibility, pair compatibility, dependency dicts
+    that are used to generate badge images and badge target pages.
+    """
+    self_compat_res = badge_utils._build_default_result(
+        status='UNKNOWN', details={})
+    google_compat_res = badge_utils._build_default_result(
+        status='UNKNOWN', details={})
+    dependency_res = badge_utils._build_default_result(
+        status='UNKNOWN', include_pyversion=False, details={})
 
-    return flask.render_template(
-        'one-badge.html',
-        package_name=package_name,
-        self_compat_res=self_compat_res,
-        google_compat_res=google_compat_res,
-        dependency_res=dependency_res,
-        commit_number=commit_number)
+    # If a package is not whitelisted, return defaults
+    if not compat_utils._is_package_in_whitelist([package_name]):
```

Comment:
Shouldn't this check be don't first?

Reply:
I think you have a typo in this comment - please clarify?

Reply:
Summary of 1:1 conversation:
Everything above this line defines default values to return if the statement is true; everything below would do the actual calculations.